### ISSUE TITLE
[cpd] Text renderer is misleading as to match start / end

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#4725]
Hi, I have called PMD with the correct parameters（pmd-bin-7.0.0-rc3 and pmd-bin-6.55.0）, and it's excited that I got the expected result! It's really great!

I tried to reflect on why I did not find it was caused by my wrong use of parameters in time. I did not add any redundant parameters at the beginning, but I still seemed to encounter this problem. And then I found that **one reason was that I was misleading by the output.**

The first time I use CPD, the command is:
```bash
 \pmd-bin-7.0.0-rc3\bin> .\pmd.bat cpd  --minimum-tokens 45 --dir  "...\example"
```
the test file is like:
```java
public class Example{
    public void theFirstMethod(List<String> strList) {
        for (String str : strList) {
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
        }
    }

    public void theSecondMethod(List<String> strList) {
        for (String str : strList) {
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
        }
    }
}
```
and the result is:
```bash
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/Users.../pmd-bin-7.0.0-rc3/lib/groovy-2.4.21.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[main] INFO net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand - Log level is at INFO
Found a 9 line (58 tokens) duplication in the following files:
Starting at line 2 of ...\example\Example.java
Starting at line 12 of...\example\Example.java

    public void theFirstMethod(List<String> strList) {
        for (String str : strList) {
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
        }
    }
```
As we can see, the source codes on line 2(method name is `theFirstMethod`) and line 12(method name is `theSecondMethod`) are different, and the output code ` public void theFirstMethod` should not appear. So I begin to try args like `--ignore-identifiers false`、 `--ignore-identifiers true`, and of cource it does not work.

Now I realize it's because the duplicates starts in one of the columns of the starting line. If I use`--format xml`, it will be show:

```bash
...\pmd-bin-7.0.0-rc3\bin> .\pmd.bat cpd  --minimum-tokens 45 --dir  "...\example" --format xml
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:.../pmd-bin-7.0.0-rc3/lib/groovy-2.4.21.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[main] INFO net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand - Log level is at INFO
<?xml version="1.0" encoding="UTF-8"?>
<pmd-cpd>
   <file path="...\example\Example.java"
         totalNumberOfTokens="127"/>
   <duplication lines="9" tokens="58">
      <file begintoken="7"
            column="31"
            endcolumn="6"
            endline="10"
            endtoken="64"
            line="2"
            path="...\example\Example.java"/>
      <file begintoken="68"
            column="32"
            endcolumn="6"
            endline="20"
            endtoken="125"
            line="12"
            path="...\example\Example.java"/>
      <codefragment><![CDATA[    public void theFirstMethod(List<String> strList) {
        for (String str : strList) {
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
        }
    }]]></codefragment>
   </duplication>
</pmd-cpd>
```

the column has been shown. However, the codefragment still have `public void theFirstMethod` which belongs to different content.

**How about when the output is accurate to line（with no column）, if the repeating code starts at the beginning of the line, the beginning line counts as this line, and when the repeating code starts in the middle of the line, the beginning line counts as the next line?**

For example, in this case, when the command is
```bash
 \pmd-bin-7.0.0-rc3\bin> .\pmd.bat cpd  --minimum-tokens 45 --dir  "...\example"
````
the result is:
```bash
Starting at line 3 of ...\example\Example.java
Starting at line 13 of...\example\Example.java
        for (String str : strList) {
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
            System.out.println(str);
        }
    }
```

Best wishes!

_Originally posted by @Scrates1 in https://github.com/pmd/pmd/issues/4676#issuecomment-1732213990_

**Comment by oowekyala** [comment]
I'm not sure this makes sense. A significant part of the first line is part of the duplicate: `(List<String> strList) {`
Also this helps provide context for the duplicate.

**Comment by jsotuyod** [comment]
Yeah, I wouldn't skip the start line either, but maybe changing the formatting to be clear where the match starts, by either adding some chars to delimit it (ie: "<<<" and ">>>") or color if rendering to console.

I do agree than info gets lost in this format otherwise.
